lib/logstash/filters/fingerprint.rb

```diff
@@ -104,6 +104,21 @@ class << self; alias_method :fingerprint, :fingerprint_openssl; end
     end
   end
 
+  def serialize(event)
+    to_string = ""
+    if event.respond_to?(:to_hash)
+      to_string << "{"
+      event.to_hash.sort.map do |k,v|
+        to_string << "#{k}:#{serialize(v)},"
```

Comment:
this would be a breaking change in two ways:
 - current `Hash#to_s` outputs key/value pairs joined by `=>`, not by `:`
 - ordering